system/Validation/FormatRules.php

```diff
@@ -344,6 +344,11 @@ public function valid_date(?string $str = null, ?string $format = null): bool
         $date   = DateTime::createFromFormat($format, $str);
         $errors = DateTime::getLastErrors();
 
+        // PHP 8.2 or later.
+        if ($date !== false && $errors === false) {
```

Comment:
`$date === false` or `! $date instanceof DateTime` should can be returned early

Reply:
Done!